gridstatus/ercot_api/ercot_api.py

```diff
@@ -1,15 +1,295 @@
 import argparse
+import io
+import os
+import time
+import zipfile
 from typing import Optional
 
 import pandas as pd
 import requests
 from tqdm import tqdm
 
+from gridstatus.decorators import support_date_range
+from gridstatus.ercot import Ercot
 from gridstatus.ercot_api.api_parser import get_endpoints_map
 from gridstatus.gs_logging import log
 
+TOKEN_URL = "https://ercotb2c.b2clogin.com/ercotb2c.onmicrosoft.com/B2C_1_PUBAPI-ROPC-FLOW/oauth2/v2.0/token"  # noqa
 BASE_URL = "https://api.ercot.com/api/public-reports"
 
+# https://data.ercot.com/data-product-archive/NP4-183-CD
+DAM_LMP_HOURLY_EMIL_ID = "NP4-183-CD"
+
+TOKEN_EXPIRATION_SECONDS = 3600
+
+
+class ErcotAPI:
+    """
+    Class to authenticate and make requests to the ERCOT Data API (api.ercot.com)
+
+    WARNING: the API appears to be a WIP and may change without notice.
```

Comment:
The API feels incomplete. You can't filter on locations or easily by dates/times and the files are missing metadata. It works for this dataset with some assumptions (`postedDate` is the same as the date for the file) but might not be ready for moving over other datasets.

Reply:
Also, the [API specification for the DAM Hourly LMP](https://github.com/ercot/api-specs/blob/378c6a04551a7095d41903d7009afd051b08f0c6/pubapi/pubapi-apim-api.json#L9506-L9723) does not match what is returned.